common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -839,9 +852,19 @@ def _cache_children(self, course_key, items, depth=0):
 
         return data
 
-    def _load_item(self, course_key, item, data_cache, apply_cached_metadata=True):
+    def _load_item(self, course_key, item, data_cache, apply_cached_metadata=True, using_descriptor_system=None):
         """
         Load an XModuleDescriptor from item, using the children stored in data_cache
+
+        Arguments:
+            course_key (CourseKey): which course to load from
+            item (UsageKey): which xblock to load
+            data_cache (dict): A dictionary mapping from UsageKeys to xblock field data
+                (this is the xblock data loaded from the database)
+            apply_cached_metadata (bool): Whether to use the cached metadata for inheritance
+                purposes.
+            using_descriptor_system (CachingDescriptorSystem): The existing CachingDescriptorSystem
+                to add data to, and to load the XBlocks from.
```

Comment:
Thanks! Maybe a contract? Though - stepping though contract code in the debugger is beginning to annoy me greatly...